src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Divide.cs

```diff
@@ -70,11 +72,80 @@ public static void Divide<T>(T x, ReadOnlySpan<T> y, Span<T> destination)
         internal readonly struct DivideOperator<T> : IBinaryOperator<T> where T : IDivisionOperators<T, T, T>
         {
             public static bool Vectorizable => typeof(T) == typeof(float)
-                                            || typeof(T) == typeof(double);
+                                            || typeof(T) == typeof(double)
+#if NET10_0_OR_GREATER
+                                            || ((Avx512F.IsSupported || Avx.IsSupported) && typeof(T) == typeof(int));
+#else
+                ;
+#endif
             public static T Invoke(T x, T y) => x / y;
-            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y) => x / y;
-            public static Vector256<T> Invoke(Vector256<T> x, Vector256<T> y) => x / y;
-            public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y) => x / y;
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
+            {
+#if NET10_0_OR_GREATER
+                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+                {
+                    return x / y;
+                }
+
+                Debug.Assert(Avx.IsSupported && typeof(T) == typeof(int));
+                Vector128<int> denominator_zero = Sse2.CompareEqual(y.AsInt32(), Vector128<int>.Zero);
```

Comment:
All of these shouldn't be written in x86 intrinsics. Comparison, conversion and divide are all having cross platform intrinsic available.

Reply:
Comparison and divide have appropriate cross platform instrinsics, but the conversion uses x86 intrinsics that convert `Int32 -> Double`. I did not see a cross platform convert that supports that case, but maybe I missed it? If there is a cross platform convert that supports the conversion, I'm happy to switch to that.

Reply:
It would be for `ConvertToDouble(WidenLower(vectorOfInt32))` which we don't currently optimize but would like to pattern match and have implicitly emit the more optimal `Sse2.ConvertToVector128Double` API

Correspondingly we'd want something like `Create(ConvertToDouble(WidenLower(vectorOfInt32)), ConvertToDouble(WidenUpper(vectorOfInt32)))` or one of the other patterns to work for `Avx.ConvertToVector256Double`

I think its fine to write the xplat code here first and then log the issue for optimizing it, which should also help prioritize getting that optimization implemented and ensure that it lights up on more hardware by default.